library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -167,13 +170,11 @@ public DownloadBatch retrieveBatchFor(DownloadInfo download) {
     }
 
     public void deleteMarkedBatchesFor(Collection<DownloadInfo> downloads) {
-        Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, null, SELECT_MARKED_FOR_DELETION, null, null);
+        Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, PROJECT_BATCH_ID, SELECT_DELETED, WHERE_MARKED_FOR_DELETION, null);
```

Comment:
:x: You tests should fail. The `?` is missing.

Furthermore, the third argument is the where clause not the fourth.
WHERE_MARKED_FOR_DELETION should be DELETION_VALUE or DELETION_FLAG in contrast to KEEP_VALUE or KEEP_FLAG

If you want to go with the "selection argument" then maybe something like:
`Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, PROJECT_BATCH_ID, WHERE_DELETE_VALUE_IS, DELETION_FLAG, null);`